### PR TITLE
Add a script we can run to backfill any missing nft mints

### PR DESCRIPTION
**Pull request by ydruffin-cb**
closes https://github.com/Stand-With-Crypto/swc-web/issues/653

**What changed? Why?**
Added an inngest function to backfill NFTs because the functionality was turned off and we need to mint all the NFT of the actions performed during that time.

**Notes to reviewers**
Please pay attention to the Prisma request in src/bin/oneTimeScripts/backfillNFT.ts to be sure that I got it right.

**How has it been tested?**

- [x] Locally
- [x] Vercel Preview Branch

**Change management**
type=routine 
risk=low 
impact=sev5

### CODE REVIEW

File: src/inngest/functions/backfillNT.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { backfillNFT } from '@/bin/oneTimeScripts/backfillNFT'
+import { inngest } from '@/inngest/inngest'
+import { onScriptFailure } from '@/inngest/onScriptFaillure'
```
Comment (benson-budiman-cb) [inline src/inngest/functions/backfillNT.ts:3]: nit: there's also a type here - `onScriptFaillure`

File: src/inngest/functions/backfillNT.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  async ({ event, step }) => {
+    const payload = event.data as ScriptPayload
+    return await step.run('execute-script', async () => {
+      return await backfillNFT({
+        limit: payload.limit,
+        persist: payload.persist,
+      })
+    })
+  },
+)
```
Comment (benson-budiman-cb) [inline src/inngest/functions/backfillNT.ts:28]: nit: this filename has a typo - `backfillNT`

File: src/bin/oneTimeScripts/backfillNFT.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const logger = getLogger('backfillNFT')
+
+export const zodBackfillNFParameters = object({
+  limit: number().optional(),
+  persist: boolean().optional(),
+})
+
+const GO_LIVE_DATE = new Date('2024-02-25 00:00:00.000')
+const ACTION_WITH_NFT: UserActionType[] = [
```
Comment (travisbloom-cb) [inline src/bin/oneTimeScripts/backfillNFT.ts:16]: We should use [this logic](https://github.com/Stand-With-Crypto/swc-web/blob/main/src/utils/server/nft/claimNFT.ts#L27) to determine what Actions have nfts

File: src/bin/oneTimeScripts/backfillNFT.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  UserActionType.VOTER_REGISTRATION,
+]
+
+interface BackfillNFTResponse {
+  dryRun: boolean
+  actionFound: number
+  airdropRequested: number
+}
+
+export async function backfillNFT(parameters: z.infer<typeof zodBackfillNFParameters>) {
```
Comment (travisbloom-cb) [inline src/bin/oneTimeScripts/backfillNFT.ts:29]: We prob want fns like this (and the last one) to not be in bin because they arent bin files. Maybe you create`src/inngest/functions/backfillNT/logic.tsx` for this stuff and the actual inggest fn is `src/inngest/functions/backfillNT/index.tsx`

File: src/bin/oneTimeScripts/backfillNFT.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    return {
+      dryRun: true,
+      actionFound: userActions.length,
+      airdropRequested: 0,
+    } as BackfillNFTResponse
+  }
+
+  const promises = []
+  for (const userAction of userActions) {
+    promises.push(claimNFT(userAction, userAction.user.primaryUserCryptoAddress!, true))
```
Comment (travisbloom-cb) [inline src/bin/oneTimeScripts/backfillNFT.ts:61]: we need to batch these prob https://github.com/Stand-With-Crypto/swc-web/blob/main/src/utils/shared/batchAsyncAndLog.ts#L5

File: src/utils/server/nft/claimNFT.ts [hunk 6]
```diff
@@ -54,12 +54,17 @@ export const ACTION_NFT_SLUG: Record<
 
 const logger = getLogger('claimNft')
 
-export async function claimNFT(userAction: UserAction, userCryptoAddress: UserCryptoAddress) {
-  if (TURN_OFF_NFT_MINT) {
+export async function claimNFT(
+  userAction: UserAction,
+  userCryptoAddress: UserCryptoAddress,
+  ignoreTurnOffNFTMintFlag: boolean = false,
```
Comment (travisbloom-cb) [inline src/utils/server/nft/claimNFT.ts:60]: lets use a config arg here so its clear

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

File: src/bin/oneTimeScripts/backfillNFT.ts (hunk 3 above)
Comment (ydruffin-cb) [reply src/bin/oneTimeScripts/backfillNFT.ts:16]: haha, I tried, but I had some difficulties to get a UserActionType from that, 
Ok let me try again

File: src/bin/oneTimeScripts/backfillNFT.ts (hunk 4 above)
Comment (ydruffin-cb) [reply src/bin/oneTimeScripts/backfillNFT.ts:29]: tsx ?
I guess you meant .ts

File: src/bin/oneTimeScripts/backfillNFT.ts (hunk 5 above)
Comment (ydruffin-cb) [reply src/bin/oneTimeScripts/backfillNFT.ts]: ✔️

File: src/bin/oneTimeScripts/backfillNFT.ts (hunk 4 above)
Comment (ydruffin-cb) [reply src/bin/oneTimeScripts/backfillNFT.ts]: ✔️

File: src/bin/oneTimeScripts/backfillNFT.ts (hunk 3 above)
Comment (ydruffin-cb) [reply src/bin/oneTimeScripts/backfillNFT.ts:16]: ✔️

File: src/utils/server/nft/claimNFT.ts (hunk 6 above)
Comment (ydruffin-cb) [reply src/utils/server/nft/claimNFT.ts]: Ok I did add a config arg, not sure it is what you had in mind, let me know
